### ISSUE TITLE
Raise on warnings in test suite

### ISSUE THREAD

**Issue by mrocklin** [dask/dask#4142]
We should consider raising exceptions in our test suite if we get any warnings.  This will help us to stay up to date as upstream dependencies evolve.

It sounds like this can be done library by library, so we can do this incrementally.

**Comment by mrocklin** [comment]
@TomAugspurger you mentioned that you had converted warnings to exceptions in the dask-ml test suite just for Scikit-Learn.  I quickly grepped through dask-ml but couldn't find how you did this.  Could you post here with a snippet that achieves this?

**Comment by TomAugspurger** [comment]
That's in the `setup.cfg`

```
[tool:pytest]
xfail_strict = true
filterwarnings =
     error:::sklearn[.*]
```

Any warning coming from scikit-learn should be elevated to errors.

According to https://docs.pytest.org/en/latest/warnings.html#pytest-mark-filterwarnings, we can override that on a per-test basis with `@pytest.mark.filterwarnings('ignore...')`  if we want to override the global setting.

**Comment by jakirkham** [comment]
Is this still something we are interested in?

Meta-comment: I was tempted to label this as `maintenance`, but did not see such a label currently so went with `core`.

**Comment by mrocklin** [comment]
+1 to using core here.  I think that I (and perhaps a few others) will be happier if the number of labels we have is small.

When I did this warning->error process before with Pandas I found it valuable.  I found things that we were ignoring that did need to be resolved.

**Comment by mrocklin** [comment]
This issue is a bit open ended though, which is a slight concern.  I figure that after we get a few libraries solidified (numpy, pandas, scikit-learn?) maybe we close this out.

**Comment by TomAugspurger** [comment]
I'll look through the current warnings and get this closed this week.

**Comment by jrbourbeau** [comment]
@TomAugspurger can this be closed now that #4916 is in `master`?

**Comment by TomAugspurger** [comment]
Yep, thanks.
